### PR TITLE
[ci]: Support code diff coverage

### PR DESCRIPTION
**Pull request by xumia**
Support code diff coverage
1. Support to add the pull request checks for Code diff coverage in GitHub.
2. Support to display the code diff coverage html report in Azure Pipelines.

### CODE REVIEW

File: .azure-pipelines/build-template.yml [hunk 1]
```diff
@@ -41,6 +41,10 @@ jobs:
 - job:
   displayName: ${{ parameters.arch }}
   timeoutInMinutes: ${{ parameters.timeout }}
+  variables:
+    DIFF_COVER_CHECK_THRESHOLD: 0
```
Comment (qiluo-msft) [inline .azure-pipelines/build-template.yml:45]: Do you have some document about the two variables? What is the value range, and how do we choose the value?

**Comment by kcudnik** [comment]
how this entire feature should work ? is already framework for support that build in azpipeline and sairedis azpipeline configuration?

File: .azure-pipelines/build-template.yml (hunk 1 above)
Comment (xumia) [reply .azure-pipelines/build-template.yml:45]: @qiluo-msft , add the doc https://github.com/Azure/sonic-build-web/tree/main/code-diff-coverage

**Comment by xumia** [comment]
@kcudnik , I added an azure devops extension to support it, see https://github.com/Azure/sonic-build-web/tree/main/code-diff-coverage

File: .azure-pipelines/build-template.yml [hunk 2]
```diff
@@ -41,6 +41,10 @@ jobs:
...
+    ${{ if ne(parameters.run_unit_test, false) }}:
+      DIFF_COVER_DISABLE: 'false'
```
Comment (qiluo-msft) [inline .azure-pipelines/build-template.yml:47]: >DIFF_COVER_DISABLE [](http://example.com/codeflow?start=6&length=18)

Negative name is harder to read than postiive one. Suggest `DIFF_COVER_ENABLED` with default value `true`. Not sure how difficult for you to change code and redeploy.

**Review by qiluo-msft — APPROVED** [review]

File: .azure-pipelines/build-template.yml (hunk 2 above)
Comment (xumia) [reply .azure-pipelines/build-template.yml:47]: Maybe some day we can change the default value to false, so no one need to set the value. The feature is in preview status, we do not want to have any impact, so disable it by default.

Comment (qiluo-msft) [reply .azure-pipelines/build-template.yml:47]: I am not asking to change the semantics of default value, only the name. The negative default value is actually same as your intention.

**Review by kcudnik — APPROVED** [review]

File: .azure-pipelines/build-template.yml (hunk 2 above)
Comment (xumia) [reply .azure-pipelines/build-template.yml]: Fixed, change to use DIFF_COVER_ENABLE=true.

**Review by kcudnik — APPROVED** [review]

**Review by qiluo-msft — APPROVED** [review]

**Comment by xumia** [comment]
@kcudnik , can we merge the code, there is one failure check, looks like it is not relative to the change.

**Comment by kcudnik** [comment]
your diff coverage plugin broken full coverage stats file by file:
https://dev.azure.com/mssonic/build/_build/results?buildId=39429&view=results

clicking you can't see any more how much covered is each added file , please revert your change or make it possible to view both results

**Pull request merged**